docs/user_docs/concepts/package-and-module.md

```diff
@@ -0,0 +1,154 @@
+# KCL Module and Package
+
+This section mainly describes how to organize files in KCL.
+
+## Overview
+
+Within a **module**, KCL organizes files grouped by **package**. A package can be defined within the module or externally (through KCL package manager `kpm`). In the latter case, KCL maintains a copy of the package within the module in a dedicated location.
+
+## Module
+
+A KCL module contains a configuration laid out in a directory hierarchy. It contains everything that is needed to deterministically determine the outcome of a KCL configuration. The root of this directory is marked by containing a `kcl.mod` directory. The contents of this directory are mostly managed by the kcl tool such as `kpm`, etc. In that sense, `kcl.mod` is analogous to the `.git` directory marking the root directory of a repo, but where its contents are mostly managed by the git tool. Besides, a KCL module is the largest unit of the file organization, has a fixed location of all KCL files and dependencies.
+
+> Note: The use of a KCL module e.g., `kcl.mod` is optional, but required if one wants to manage, distribute, share and reuse code with a semantic version.
+
+### Creating a module
+
+A module can be created by running the following command within the module root:
+
+```bash
+kpm init [module name]
+```
+
+The module name is **required** if a package within the module needs to import another package within the module. A module can also be created by setting up the `kcl.mod` file manually.
+
+## Package
+
+In KCL, a package is usually composed of a "folder" containing KCL files. This folder can be a real disk physical path, or it can be composed of multiple KCL files (usually main package). Different packages are uniquely located by different package paths (such as `kubernetes.core.v1`)
+
+Within the same module, different packages can be imported from each other through the import statement of relative or absolute path. In the KCL, the relative import will be replaced by absolute import and the corresponding KCL code will be found through the package path.
+
+### Relative Import Path
+
+We can use the operator `.` to realize the relative path import of KCL entry files.
+
+main.k:
+
+```python
+import .model1  # Current directory module
+import ..service  # Parent directory
+import ...root  # Parent of parent directory
+
+s = service.ImageService {}
+m = root.Schema {}
+```
+
+### Absolute Import Path
+
+The semantics of `import a.b.c.d` is
+
+1. If `kcl.mod` not exist, search the path `./a/b/c/d` from the current directory.
+2. If the current directory search fails, search from the root path `ROOT_PATH/a/b/c/d`, else raise a import error.
```

Comment:
In my local KCL enviorment(kclvm version is 0.4.5; checksum: c5339e572207211e46477825e8aca903), If the `kcl.mod` file exists, the `a.b.c` import path will be recognized as an absolute import prior to a relative import.